kubeflow/kpt-set.sh

```diff
@@ -1,22 +1,58 @@
-
-# Use kpt to set kustomize values
+# Set kpt setters using env vars.
 
 set -ex
 
-kpt cfg set -R .  gke.private false
-kpt cfg set -R .  asm-label ${ASM_LABEL}
+if [[ -z "${KF_NAME}" ]]; then
+    echo "KF_NAME env var is required"
+fi
+if [[ -z "${MGMT_NAME}" ]]; then
+    echo "MGMT_NAME env var is required"
+fi
+if [[ -z "${KF_PROJECT}" ]]; then
+    echo "KF_PROJECT env var is required"
+fi
+if [[ -z "${KF_PROJECT_NUMBER}" ]]; then
+    echo "KF_PROJECT_NUMBER env var is required"
+fi
+if [[ -z "${LOCATION}" ]]; then
+    echo "LOCATION env var is required"
+fi
+if [[ -z "${ZONE}" ]]; then
+    echo "ZONE env var is required"
+fi
+if [[ -z "${REGION}" ]]; then
+    echo "REGION env var is required"
+fi
+if [[ -z "${ADMIN_EMAIL}" ]]; then
+    echo "ADMIN_EMAIL env var is required"
+fi
+if [[ -z "${ASM_LABEL}" ]]; then
+    echo "ASM_LABEL env var is required"
+fi
 
-kpt cfg set -R .  mgmt-ctxt ${MGMT_NAME}
+# kpt setter documentation:
+# https://googlecontainertools.github.io/kpt/guides/consumer/set/
+kpt cfg set -R .  mgmt-ctxt "${MGMT_NAME}"
 
-kpt cfg set -R .  name ${KF_NAME}
-kpt cfg set -R .  gcloud.project.projectNumber ${KF_PROJECT_NUMBER}
-kpt cfg set -R .  gcloud.core.project ${KF_PROJECT}
-kpt cfg set -R .  gcloud.compute.zone ${LOCATION}
-kpt cfg set -R .  location ${LOCATION}
-kpt cfg set -R .  log-firewalls false
+kpt cfg set -R .  name "${KF_NAME}"
+kpt cfg set -R .  gcloud.core.project "${KF_PROJECT}"
+kpt cfg set -R .  gcloud.project.projectNumber "${KF_PROJECT_NUMBER}"
+kpt cfg set -R .  location "${LOCATION}"
+kpt cfg set -R .  gcloud.compute.zone "${ZONE}"
+kpt cfg set -R .  gcloud.compute.region "${REGION}"
+kpt cfg set -R .  email "${ADMIN_EMAIL}"
 
-kpt cfg set -R .  email ${ADMIN_EMAIL}
+kpt cfg set -R .  asm-label "${ASM_LABEL}"
 
-# Default values for Kubeflow Pipelines, you can override as you like.
-kpt cfg set apps/pipelines cloudsql-name "${KF_NAME}-kfp"
-kpt cfg set apps/pipelines bucket-name "${KF_NAME}-kfp-artifacts"
+# Default values for managed storage used by Kubeflow Pipelines (KFP), you can
+# override as you like.
+# The CloudSQL instance and Cloud Storage bucket instance are created during
+# deployment, so you should make sure their names are not used before.
+CLOUDSQL_NAME="${KF_NAME}-kfp"
+BUCKET_NAME="${KF_PROJECT}-kfp"
+# common/managed-storage deploys specified CloudSQL and Cloud Storage bucket.
+kpt cfg set common/managed-storage cloudsql-name "${CLOUDSQL_NAME}"
+kpt cfg set common/managed-storage bucket-name "${BUCKET_NAME}"
+# apps/pipelines uses specified CloudSQL and Cloud Storage bucket.
+kpt cfg set apps/pipelines cloudsql-name "${CLOUDSQL_NAME}"
+kpt cfg set apps/pipelines bucket-name "${BUCKET_NAME}"
```

Comment:
NIT: Note that we currently paste the whole content of `env.sh` and `kpt-set.sh` in https://www.kubeflow.org/docs/distributions/gke/deploy/deploy-cli/#environment-variables.

I can foresee that it becomes harder and harder to keep shell script, README and kubeflow.org documentation in sync when this file grows. I recommend using this file as the source of truth by documentation, to prevent out-of-sync.

Reply:
I think that's a reasonable choice

Reply:
Thank you Yuan for confirming! I will update documentation accordingly